fedimint-server/src/config/api.rs

```diff
@@ -765,6 +803,23 @@ mod tests {
                 })
                 .await
         }
+
+        /// Helper for getting server status
+        async fn server_status(&self) -> ServerStatus {
+            while self.client.status().await.is_err() {
```

Comment:
This is reading it twice, with a possibility of second one panicking.

```rust
loop {
match self.client.status().await {
  Ok(o) => return Ok(o);
  Err(e) => /* print, sleep */
}
```

?

Reply:
True, main reason the polling is needed is because we are waiting for the API to come online so it shouldn't fail the second time, but I can rewrite it better.